sdk/src/types/block/capabilities.rs

```diff
@@ -0,0 +1,207 @@
+    pub fn set_capabilities(&mut self, flags: impl IntoIterator<Item = Flag>) -> &mut Self {
+        *self = Self::default().with_capabilities(flags);
+        self
+    }
+
+    /// Returns whether a given flag is enabled.
+    pub fn has_capability(&self, flag: Flag) -> bool {
+        self.get(flag.index())
+            .map(|byte| byte & flag.as_byte() == flag.as_byte())
+            .unwrap_or_default()
+    }
+
+    /// Returns whether a given set of flags are enabled.
+    pub fn has_capabilities(&self, flags: impl IntoIterator<Item = Flag>) -> bool {
+        flags.into_iter().all(|flag| self.has_capability(flag))
+    }
+
+    /// Returns an iterator over all enabled flags.
+    pub fn capabilities_iter(&self) -> impl Iterator<Item = Flag> + '_ {
+        self.iter().enumerate().flat_map(|(idx, byte)| {
+            Flag::all().filter(move |f| (idx == f.index() && byte & f.as_byte() == f.as_byte()))
+        })
+    }
+}
+
+impl<Flag> Default for Capabilities<Flag> {
+    fn default() -> Self {
+        Self {
+            bytes: Default::default(),
+            _flag: PhantomData,
+        }
+    }
+}
+
+impl<Flag> Clone for Capabilities<Flag> {
+    fn clone(&self) -> Self {
+        Self {
+            bytes: self.bytes.clone(),
+            _flag: PhantomData,
+        }
+    }
+}
+
+impl<Flag> PartialEq for Capabilities<Flag> {
+    fn eq(&self, other: &Self) -> bool {
+        self.bytes == other.bytes
+    }
+}
+
+impl<Flag> Eq for Capabilities<Flag> {}
+
+impl<Flag> PartialOrd for Capabilities<Flag> {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        self.bytes.partial_cmp(&other.bytes)
+    }
+}
+
+impl<Flag> Ord for Capabilities<Flag> {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.bytes.cmp(&other.bytes)
+    }
+}
+
+impl<Flag> core::hash::Hash for Capabilities<Flag> {
+    fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
+        self.bytes.hash(state);
+    }
+}
+
+impl<I: IntoIterator<Item = Flag>, Flag: CapabilityFlag> From<I> for Capabilities<Flag> {
+    fn from(value: I) -> Self {
+        Self::default().with_capabilities(value)
+    }
+}
+
+impl<Flag: 'static> Packable for Capabilities<Flag> {
+    type UnpackError = crate::types::block::Error;
+    type UnpackVisitor = ();
+
+    fn pack<P: packable::packer::Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
+        if !self.is_none() {
+            self.bytes.pack(packer)?;
+        } else {
+            0_u8.pack(packer)?;
+        }
+        Ok(())
+    }
+
+    fn unpack<U: packable::unpacker::Unpacker, const VERIFY: bool>(
+        unpacker: &mut U,
+        visitor: &Self::UnpackVisitor,
+    ) -> Result<Self, packable::error::UnpackError<Self::UnpackError, U::Error>> {
+        use packable::prefix::UnpackPrefixError;
+        Ok(Self::from_bytes(
+            BoxedSlicePrefix::unpack::<_, VERIFY>(unpacker, visitor)
+                // TODO: not sure if this is the best way to do this
```

Comment:
I've been trying to figure out something for this but I can't find anything, maybe we just remove the TODO ?